pkg/containers/runtime/runtime.go

```diff
@@ -0,0 +1,44 @@
+package runtime
+
+import (
+	"context"
+	"os"
+)
+
+type ContainerMetadata struct {
+	ContainerId string
+	Name        string
+	Image       string
+}
+
+type ContainerEnricher interface {
+	Get(containerId string, ctx context.Context) (ContainerMetadata, error)
+}
+
+type Sockets struct {
+	sockets map[string]string
+}
+
+func (s *Sockets) Register(runtime string, socket string) error {
+	if s.sockets == nil {
+		s.sockets = make(map[string]string)
+	}
+
+	_, err := os.Stat(socket)
+	if err == nil {
+		s.sockets[runtime] = socket
+	}
+
+	return err
```

Comment:
The error will be returned correctly here, if it's nil the register will happen and nil will be returned, otherwise we return the os.Stat error and not do anything.
However I think adding the "invalid socket" would be better so i'll do it like this.

Reply:
done